### ISSUE TITLE
Null Reference Exception when Patching Methods with Return Type Point

### ISSUE THREAD

**Issue by danvolchek** [pardeike/Harmony#159]
Hi there,

I've been trying to patch a method with signature `private Microsoft.Xna.Framework.Point getKitchenStandingSpot()`  in the game Stardew Valley. This results in the following error:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at getKitchenStandingSpot_Patch1(Object )
```

I did some investigation and found the following things:
 - The error occurs in both Prefix/Postfix patches.
 - I get the same error regardless of whether the signature is private or public.
 - The code inside the patch itself runs fine, but then the error happens after the patch code returns (even if the patch method is empty).
 - The error does NOT occur and it works fine if the method is `static`.
 - The error does NOT occur and it works fine for a return type of `Microsoft.Xna.FrameWork.Rectangle` (which is a very similar type, I think).

This has led me to believe that the Point type is the problem, but I'm not 100% confident and not sure why it would be causing issues. Would you be able to take a look into the problem?

I'm not sure if you own Stardew Valley, but I can provide a simple mod + save file and the steps to reproduce the issue if needed. I'm able to provide any more info if needed - please ask.

By the way, Harmony is great and it lets a lot of SDV mods do things that weren't originally possible. Thanks for making it :)

**Comment by danvolchek** [comment]
I made a simple console application that performs the same patch (on a method that returns a Point) and there were no errors. So it looks like it's specifically related to Stardew Valley somehow.

**Comment by pardeike** [comment]
There is an (now closed) issue regarding methods returning structs of a specific size and the way the runtime changes the assembler calling conventions. I implemented a fix in the upcoming 2.0 version of Harmony. 

I assume you are using 1.2.0.1 so https://github.com/pardeike/Harmony/issues/77 applies to you.

**Comment by danvolchek** [comment]
I am using 1.2.0.1.

Apologies, I should have gone through the closed issues before opening up another one. Thanks for the quick followup!
